preprocess.py

```diff
@@ -40,18 +41,30 @@ def get_parser():
     parser.add_argument('--only-source', action='store_true', help='Only process the source language')
     parser.add_argument('--padding-factor', metavar='N', default=8, type=int,
                         help='Pad dictionary size to be multiple of N')
+    parser.add_argument('--tokenize_number', metavar='N', default=0, type=int, choices=range(2),
+                        help="Which tokenization algorithm to use. 0 is tokenize_line, 1 is nltk's word tokenize. Default is 0.")
+    parser.add_argument('--max_source_length', metavar='N', default=-1, type=int,
+                        help="The maximum length of the source. If a sequence is longer it will be truncated to this length. " 
+                             "If -1, then no truncation occurs. For joined_dictionary's the maximum of the source/target max length will"
+                             " be used to build the dictionary. Default is no truncation.")
+    parser.add_argument('--max_target_length', metavar='N', default=-1, type=int,
```

Comment:
can you just remove the defaults and check if max_source_length and max_target_length are None instead (instead of -1)